components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/internal/ServiceReferenceHolder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.authenticator.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.rest.api.authenticator.configuration.models.APIMStoreConfigurations;
+import org.wso2.carbon.kernel.configprovider.CarbonConfigurationException;
+import org.wso2.carbon.kernel.configprovider.ConfigProvider;
+
+import java.util.Map;
+
+/**
+ * Class used to hold the APIM store configurations.
+ */
+public class ServiceReferenceHolder {
+    private static final Logger log = LoggerFactory.getLogger(ServiceReferenceHolder.class);
+    private static ServiceReferenceHolder instance = new ServiceReferenceHolder();
+    private ConfigProvider configProvider;
+    private APIMStoreConfigurations config = null;
+
+    private ServiceReferenceHolder() {
+
+    }
+
+    public static ServiceReferenceHolder getInstance() {
+        return instance;
+    }
+
+    public void setConfigProvider(ConfigProvider configProvider) {
+        this.configProvider = configProvider;
+    }
+
+    public APIMStoreConfigurations getAPIMStoreConfiguration() {
+        try {
+            if (configProvider != null) {
+                config = configProvider.getConfigurationObject(APIMStoreConfigurations.class);
+            } else {
+                log.error("Configuration provider is null");
```

Comment:
Because configiProvider null means config is becoming null  which you then set the default configurationss